ext/opcache/tests/jit/fetch_dim_r_003.phpt

```diff
@@ -54,8 +54,8 @@ string(1) "A"
 Warning: Illegal string offset 'ab' in %sfetch_dim_r_003.php on line 15
 string(1) "A"
 
-Notice: A non well formed numeric value encountered in %sfetch_dim_r_003.php on line 16
+Warning: Illegal string offset '2x' in %sfetch_dim_r_003.php on line 16
 string(1) "C"
```

Comment:
Shouldn't this produce `"A"` now?

Reply:
Indeed it should, let me see if I can figure out the JIT.

Reply:
Actually I'm not sure, as even the non JIT case has this behaviour, best seen with the ``Zend/tests/offset_string.phpt`` test, where floats also emit this warning but get convert to an int with the allow errors mode.

So what's the plan for that? Use the strict mode for indexes?

Reply:
So the problem here seems to be that is_numeric_string is used to validate the offset, but then the actual offset is computed via an unconditional (int) cast, not via the is_numeric_string result. Not sure about this.

Reply:
I'll bring it up on the list to see what people think.